### PR TITLE
Remove `Engine`

### PR DESCRIPTION
**Pull request by nnethercote**
It's just unnecessary plumbing. Removing it results in less code, and simpler code.

r? @cjgillot

### CODE REVIEW

File: compiler/rustc_borrowck/src/lib.rs [hunk 1]
```diff
@@ -193,9 +193,7 @@ fn do_mir_borrowck<'tcx>(
         .map(|(idx, body)| (idx, MoveData::gather_moves(body, tcx, |_| true)));
 
     let mut flow_inits = MaybeInitializedPlaces::new(tcx, body, &move_data)
-        .into_engine(tcx, body)
-        .pass_name("borrowck")
-        .iterate_to_fixpoint()
+        .iterate_to_fixpoint(tcx, body, Some("borrowck"))
```
Comment (cjgillot) [inline compiler/rustc_borrowck/src/lib.rs:196]: Further simplification: should we always give a name and get rid of the option?

**Review by cjgillot — COMMENTED** [review]
R=me with one nit

File: compiler/rustc_borrowck/src/lib.rs (hunk 1 above)
Comment (nnethercote) [reply compiler/rustc_borrowck/src/lib.rs:196]: I just took a look. The optional pass name is unwrapped as `"-----"` here:

https://github.com/rust-lang/rust/blob/1e4f10ba6476e48a42a79b9f846a2d9366525b9e/compiler/rustc_mir_dataflow/src/framework/engine.rs#L215

So everywhere I've used `None` would be changed to `"-----"` which would be weird. More generally, the file naming scheme for IR dumping is complicated. I don't want to delve into that for this PR, which is a straightforward refactoring. Maybe in a follow-up.

**Comment by nnethercote** [comment]
Thanks for the fast review.

@bors r+ rollup

**Pull request merged**